recipes/async_simple/all/conandata.yml

```diff
@@ -1,7 +1,14 @@
 sources:
+  "1.3":
+    url: "https://github.com/alibaba/async_simple/archive/refs/tags/v1.3.tar.gz"
+    sha256: "0BA0DC3397882611B538D04B8EE6668B1A04CE046128599205184C598B718743"
   "1.2":
     url: "https://github.com/alibaba/async_simple/archive/refs/tags/1.2.tar.gz"
     sha256: "a59a2674ac2b0a3997b90873b2bf0fbe4d96fdedbe6a2628c16c92a65a3fa39a"
   "1.0.0":
     url: "https://github.com/alibaba/async_simple/archive/refs/tags/1.0.tar.gz"
     sha256: "b243fb7af5d61b534fe18b662d16498392bbce1deffdc68e58829aa31db7badf"
+patches:
+  "1.0.0":
+    - patch_file: "patches/disable-asan-1.0.patch"
```

Comment:
Could you please add more information about these patches? What's the origin, the motivation, type ... 

https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/conandata_yml_format.md#patches-fields

Reply:
@RazielXYZ Thank for adding extra information.

The patch https://github.com/alibaba/async_simple/commit/ccb760356b9d7481ca73f0c58d15928bf8b0aa0b does quite opposite, it add the option ASYNC_SIMPLE_ENABLE_ASAN and enables it by default. Using this patch, it will disable actually. I don't see it following the upstream default behavior, but diverging from it. Am I missing something?

Reply:
@uilianries The upstream default behavior is to enable ASAN, yes, but prior to that commit they had no option for it, and therefore no way to disable it. I assume we don't want packages built with ASAN on conancenter, as they do not work when linked against things not built with ASAN (unless I'm wrong!). As such, before they added the option, the only way to not have it build with ASAN would be to remove -fsanitize=address from the cmakefiles which these patches do. If the extra info makes that unclear I am open to suggestions!

Reply:
Agreed. Could you please use the official patch, then pass the cmake definition to disable asan (no conan option). Just to keep officially following the upstream.